From 047ab6bed083011e31681ce4d5ed629a94270507 Mon Sep 17 00:00:00 2001
From: Jonas Albrecht <plonkbong100@protonmail.com>
Date: Sun, 8 May 2022 15:04:31 +0000
Subject: [PATCH] change NAND ECC POS values

change NAND ECC Positions values for the device: arcadyan vgv953akw22-*

Signed-off-by: Jonas Albrecht <plonkbong100@protonmail.com>
---
 arch/mips/include/asm/lantiq/config.h | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/arch/mips/include/asm/lantiq/config.h b/arch/mips/include/asm/lantiq/config.h
index bcc7b61..fb15ee7 100644
--- a/arch/mips/include/asm/lantiq/config.h
+++ b/arch/mips/include/asm/lantiq/config.h
@@ -1,6 +1,7 @@
 /*
  * Copyright (C) 2007-2010 Lantiq Deutschland GmbH
  * Copyright (C) 2011-2013 Daniel Schwierzeck, daniel.schwierzeck@gmail.com
+ * Copyright (C) 2022 Jonas Albrecht, plonkbong100@protonmail.com
  *
  * SPDX-License-Identifier:	GPL-2.0+
  */
@@ -56,9 +57,16 @@
 #define CONFIG_SPL_NAND_SOFTECC
 #define CONFIG_SYS_NAND_ECCSIZE		256
 #define CONFIG_SYS_NAND_ECCBYTES	3
-#define CONFIG_SYS_NAND_ECCPOS		{40, 41, 42, 43, 44, 45, 46, 47, \
-					48, 49, 50, 51, 52, 53, 54, 55, \
-					56, 57, 58, 59, 60, 61, 62, 63}
+#define CONFIG_SYS_NAND_ECCPOS		{0, 1, 2, 3, 6, 7, \
+					40, 41, 42, 43, 44, 45, 46, 47, \
+ 					48, 49, 50, 51, 52, 53, 54, 55, \
+					56, 57, 58, 59, 60, 61, 62, 63, \
+					80, 81, 82, 83, 84, 85, 86, 87, \
+					88, 89, 90, 91, 92, 93, 94, 95, \
+					96, 97, 98, 99, 100, 101, 102, 103, \
+					104, 105, 106, 107, 108, 109, 110, 111, \
+					112, 113, 114, 115, 116, 117, 118, 119, \
+					120, 121, 122, 123, 124, 125, 126, 127}
 #endif
 
 #if defined(CONFIG_LTQ_SUPPORT_SPL_NOR_FLASH) && defined(CONFIG_SYS_BOOT_NORSPL)
-- 
2.32.0

